File: 1306-minimum-absolute-difference/1306-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> ans = new ArrayList<>();
        Arrays.sort(arr);
        int n = arr.length;

        int diff = Integer.MAX_VALUE;

        for(int i = 0; i < n-1; i++){
            diff = Math.min(diff, arr[i+1] - arr[i]);
        }

        for(int i = 0; i< n - 1; i++){
            List<Integer> each = new ArrayList<>();
            if(arr[i+1] - arr[i] == diff){
                each.add(arr[i]);
                each.add(arr[i+1]);
            }
            if(each.size()>0) ans.add(each);
        }

        return ans;
    }
}

File: 1306-minimum-absolute-difference/1306-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        vector<vector<int>> ans;

        sort(begin(arr), end(arr));
        int n = arr.size();
        int diff = arr[1] - arr[0];
        for(int i = 1; i < n-1 ; i++){
            diff = min(diff, arr[i+1] - arr[i]);
        }

        for(int i = 0; i < n-1; i++){
            vector<int> each;
            if(arr[i+1] - arr[i] == diff){
                each.push_back(arr[i]);
                each.push_back(arr[i + 1]);
            }
            if(each.size()) ans.push_back(each);
        }

        return ans;
    }
};